pages/mesosphere/dcos/2.2/storage/external-storage/csi/index.md

```diff
@@ -0,0 +1,119 @@
+  "cpus": 0.1,
+  "mem": 32,
+  "cmd": "date >> test-csi-volume/test.txt; cat test-csi-volume/test.txt",
+  "container": {
+    "type": "MESOS",
+    "volumes": [
+      {
+        "containerPath": "test-csi-volume/",
+        "external": {
+          "provider": "csi",
+          "name": "pre-provisioned-volume-id-001",
+          "pluginName": "csi-plugin-name",
+          "options": { 
+            "accessMode": "MULTI_NODE_MULTI_WRITER",
+            "nodeStageSecret": {
+              "username": "username_secret_001",
+              "password": "password_secret_001"
+            },
+            "nodePublishSecret": {
+              "username": "username_secret_001",
+              "password": "password_secret_001"
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+### Volume configuration options
+
+-  `containerPath`: The path where the volume is mounted inside the container.
+-  `external.name`: The unique ID that the CSI volume driver uses to look up your volume. When your task is staged on an agent, the volume driver queries the storage service for a volume with this name. If one does not exist, the task will fail to launch.
+-  `external.pluginName`: The name of the CSI plugin which will mount this volume. This is specified in the CSI plugin configuration on the agent; for the Portworx service, this should be `pxd.portworx.com`. For NFS volumes, this should be `nfs.csi.k8s.io`. For other drivers, see the instructions on driver installation below.
+-  `external.options.accessMode`: The [CSI access mode][2] to use when mounting the volume. The `XXXX_WRITER` modes will be mounted in read-write, while the `READER_ONLY` modes will be mounted read-only.
+-  `external.options.nodeStageSecret` and `external.options.nodePublishSecret`: The names of secrets in the DC/OS secret store which contain the username/password to use with the storage backend when staging and publishing this volume. "Staging" and "publishing" refer to steps in the CSI volume mounting process.
+-  `external.options.accessType.fsType`: The type of filesystem to be used on this volume, i.e. "xfs", "ext4", etc.
+-  `external.options.accessType.mountFlags`: Flags to be used when mounting the volume on the host.
+-  `external.options.volumeContext`: In some cases, the provisioning of a volume results in the creation of volume context metadata which must be passed as key-value pairs in this field.
+-  Create multiple volumes by adding additional items in the `container.volumes` array.
+-  You cannot change volume parameters after you create the application.
+-  Marathon will not launch apps with external volumes if `upgradeStrategy.minimumHealthCapacity` is greater than 0.5, or if `upgradeStrategy.maximumOverCapacity` does not equal 0.
+
+Some of the options above map directly onto fields in the [`NodeStageVolume`][3] and [`NodePublishVolume`][4] calls in the CSI spec; some useful information regarding those parameters can be found in the CSI docs.
```

Comment:
Can we add the link to the CSI spec here as well?  I know it is in the intro section but I think it would improve this example if you could get to from here.